mos-platform/geos-cbm/geos_constants.h

```diff
@@ -0,0 +1,752 @@
+/***********************************************************************
+ * geos_constants.h
+ *
+ * This file contains #defines for use in GEOS applications
+ * with llvm-mos C compiler.
+ *
+ * Transcribed from geoProgrammer manual by DiSc.
+ *
+ ***********************************************************************/
+
+#ifndef GEOS_CONSTANTS_H
+#define GEOS_CONSTANTS_H
+
+/***********************************************************************
+ * Miscellaneous defines
+ ***********************************************************************/
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#define LOW_BYTE(x) (*((unsigned char *)&(##x) + 0))
+#define HIGH_BYTE(x) (*((unsigned char *)&(##x) + 1))
+
+#define WORD(w) (w & 0xff), (w >> 8)
+
+#define AUX(x) #x
+#define STRINGIFY(x) AUX(x)
+
+/***********************************************************************
+ * Hardware related defines
+ ***********************************************************************/
+
+/** values for CPU_DATA - C64 memory config */
+
+#define RAM_64K 0x30        /** 64K RAM */
+#define IO_IN 0x35          /** 60K RAM, 4K I/O space in */
+#define KRNL_IO_IN 0x36     /** kernal and I/O mapped into memory */
+#define KRNL_BAS_IO_IN 0x37 /** kernal, I/O and basic mapped into memory */
+
+/** values for config - C128 mmu */
+
+#define CIOIN 0x7E        /** 60K RAM, 4K I/O space in */
+#define CRAM64K 0x7F      /** 64K RAM */
+#define CKRNLBASIOIN 0x40 /** kernal, I/O and basic ROM's mapped into memory   \
+                           */
+#define CKRNLIOIN 0x4E    /** Kernal ROM and I/O space mapped in */
+
+/***********************************************************************
+ * Menu Equates
+ ***********************************************************************/
+
+/** Menu types */
+typedef enum __attribute__((__packed__)) {
+  HORIZONTAL = 0b00000000,
+  VERTICAL = 0b10000000,
+  CONSTRAINED = 0b01000000,
+  UN_CONSTRAINED = 0b00000000
```

Comment:
`UNCONSTRAINED`? It's an English word, so it's odd to see it in snake_case. Disregard if this was taken verbatim from a more authoritative source.

Reply:
Its written that way in the GEOS original documentation.